### PR TITLE
Smoke Tests

### PR DESCRIPTION
**Pull request by maxscheurer**
Minimal testing of functionality.

### CODE REVIEW

File: adcc/smoketests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python3
```
Comment (mfherbst) [inline adcc/smoketests/__init__.py:1]: Why is this file needed at all?

File: adcc/smoketests/test_smoke.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        scfres = cached_backend_hf("pyscf", "h2o", "sto-3g")
+        refstate = adcc.ReferenceState(scfres)
+        state_singlets = adcc.run_adc(refstate, method=method,
+                                      n_singlets=3, conv_tol=1e-7)
+        state_triplets = adcc.run_adc(refstate, method=method,
+                                      n_triplets=3, conv_tol=1e-7)
+        assert_allclose(state_singlets.excitation_energy,
+                        _singlets[method], atol=1e-6)
+        assert_allclose(state_triplets.excitation_energy,
+                        _triplets[method], atol=1e-6)
```
Comment (mfherbst) [inline adcc/smoketests/test_smoke.py:80]: Comparing the residual norm of the eigenpairs did not work?

File: adcc/smoketests/test_smoke.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
```
Comment (mfherbst) [inline adcc/smoketests/test_smoke.py:1]: I would've just put this in the main folder as well or is there any reason I miss why this should be in a subfolder?

**Review by mfherbst — COMMENTED** [review]
Thanks!

File: adcc/smoketests/__init__.py (hunk 1 above)
Comment (maxscheurer) [reply adcc/smoketests/__init__.py:1]: I wanted to invoke with `pytest --pyargs adcc.smoketests`, but can as well go for `pytest --pyargs adcc`. Probably the latter version is better (= shorter)...

File: adcc/smoketests/test_smoke.py (hunk 2 above)
Comment (maxscheurer) [reply adcc/smoketests/test_smoke.py:80]: Mhmm I don't see how this could be done from the `ExcitedStates` object... am I missing something?

Comment (mfherbst) [reply adcc/smoketests/test_smoke.py:80]: can't you just compute M * v - λ * v and check the norm is small? (M = ADC matrix)

Comment (maxscheurer) [reply adcc/smoketests/test_smoke.py:80]: Yes, I can 😄

**Comment by maxscheurer** [comment]
Running the SCF manually now, because `adcc.testdata.static_data` is not installed, and I don't think we should install it 😄 Repeated H2O/sto-3g SCF does not cost that much, so it's not a problem.

File: adcc/test_smoke.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    return np.array([r @ r for r in residuals])
+
+
+@pytest.mark.skipif("pyscf" not in adcc.backends.available(),
+                    reason="PySCF not found.")
+@expand_test_templates(methods)
+class TestSmoke(unittest.TestCase):
+    conv_tol = 1e-7
+
+    def _run_scf_h2o_sto3g(self):
```
Comment (mfherbst) [inline adcc/test_smoke.py:75]: Make a comment why this and not using the cache.

File: adcc/test_smoke.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        scfres = adcc.backends.run_hf(
+            "pyscf", xyz="""
+            O 0 0 0
+            H 0 0 1.795239827225189
+            H 1.693194615993441 0 -0.599043184453037""",
+            basis="sto-3g", conv_tol=1e-12,
+            conv_tol_grad=1e-10
```
Comment (mfherbst) [inline adcc/test_smoke.py:82]: could make this depend on the class variable `conv_tol` as well.

**Review by mfherbst — APPROVED** [review]
Great!

**Comment by maxscheurer** [comment]
Drat... one cannot install pyscf via pip in the `conda-build` test environment... So we need to use hard-coded reference state data.

**Comment by maxscheurer** [comment]
BTW I just noticed that when a wheel dist is built using `python -m pip install . -vv` from adcc source directory, all the test files, including `conftest.py` are getting installed. The `MANIFEST` file is only acting on `sdist` builds, such that all our `conda` builds (which ran the above command or `python setup.py install`) include all the test files, even if they don't work in practice 😬 

Now I don't know how to **exclude** these specific files from the distribution, because the `exclude` statement in `find_packages` only acts on submodules, but not individual py files. 👎

**Comment by mfherbst** [comment]
That's annoying ... I don't know a fix either. I would say, let's try very hard to find something and otherwise ... that's life. I mean one could an explicit `rm` in the conda build stage, deleting all files but the static `test_smoke.py` ... or even better maybe have a script in the adcc sources which does this (cleanup_tests or so).

**Comment by maxscheurer** [comment]
Yes, I vote for the most pragmatic solution 😄  I hate these rabbit holes with deployment etc.
